schemas/configuration/v5/server_config.json.in

```diff
@@ -6,38 +6,89 @@
+                        }
                     }
                 },
                 "hostname_cache": {
                     "type": "object",
                     "properties": {
-                        "shared_memory_size_in_bytes": {"type": "integer"},
-                        "eviction_age_in_seconds": {"type": "integer"},
-                        "cache_clearer_sleep_time_in_seconds": {"type": "integer"}
+                        "shared_memory_size_in_bytes": {
+                            "type": "integer",
+                            "minimum": 0
+                        },
+                        "eviction_age_in_seconds": {
+                            "type": "integer",
+                            "minimum": 0
+                        },
+                        "cache_clearer_sleep_time_in_seconds": {
+                            "type": "integer",
+                            "minimum": 0
+                        }
                     }
                 },
-                "maximum_size_for_single_buffer_in_megabytes": {"type": "integer"},
-                "maximum_size_of_delay_queue_in_bytes": {"type": "integer"},
-                "maximum_temporary_password_lifetime_in_seconds": {"type": "integer"},
-                "migrate_delay_server_sleep_time_in_seconds":  {"type": "integer"},
-                "number_of_concurrent_delay_rule_executors": {"type": "integer"},
-                "stacktrace_file_processor_sleep_time_in_seconds": {"type": "integer"},
-                "transfer_buffer_size_for_parallel_transfer_in_megabytes": {"type": "integer"},
-                "transfer_chunk_size_for_parallel_transfer_in_megabytes": {"type": "integer"}
+                "maximum_size_for_single_buffer_in_megabytes": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "maximum_size_of_delay_queue_in_bytes": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "maximum_temporary_password_lifetime_in_seconds": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "migrate_delay_server_sleep_time_in_seconds": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "number_of_concurrent_delay_rule_executors": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "stacktrace_file_processor_sleep_time_in_seconds": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "transfer_buffer_size_for_parallel_transfer_in_megabytes": {
+                    "type": "integer",
+                    "minimum": 0
+                },
+                "transfer_chunk_size_for_parallel_transfer_in_megabytes": {
+                    "type": "integer",
+                    "minimum": 0
+                }
```

Comment:
Does a value of 0 for either of these effectively disable parallel transfer?

Reply:
Again, one of those things I'm unsure about.

We certainly don't appear to have any tests around it. The schema changes are something we have to inspect by looking at where the properties are used in the code.

Reply:
And so… what happens when a 0 is used?   Transfer just doesn’t happen?

Does the user and/or server get an error?  Are they blocked completely?   

Is this what we want?

Reply:
Well, a value of 0 is likely to lead to an error or undefined behavior if it makes it to the stage of being used to allocate a buffer.

The code I glanced over didn't always immediately lead to memory allocation though. Some places checked the value and rejected it if it was less than 1. Other locations didn't check the value.

A minimum value of 1 is likely more correct as 0 doesn't make a lot of sense when it comes to buffer sizes unless it's considered special and leads to well-defined behavior.

iRODS 4 doesn't enforce any of these values, so we aren't any worse than before. With that said, I think the answer is - set the minimums to 1 and see if the core tests pass. I've watched all unit, core, federation, and topology tests pass with minimums set to 0.

Reply:
agreed - proceeding with 1 is the most defensible.

Reply:
Changed minimums to 1.